### DISCUSSION TITLE
Background-color or background-image on body has no effect

### DISCUSSION THREAD

**Discussion by shimonmagal** [gotenberg/gotenberg#572] [Q&A]
Hi, I am using gotenberg to create PDFs and now I want to make them stylish.
It doesn't seem like:

`body {
   background-color: #ff0000;
}`

has any effect. Same goes with background-image.
My ultimate goal is to be able to attach a background-image on each PDF page to cover it fully.
But even simple background-color produces a regular PDF, no background.

Steps to reproduce:
1. `docker run --restart=always -p 3000:3000 gotenberg/gotenberg:7.7`
2. Creating a file index.html:
```
<html>
<style>
    body {
        background-color: "ff0000"
    }
</style>

<body>
    <p>hi</p>
    <p>hi</p>
    <p>hi</p>
</body>

</html>
```
3. Producing the file using a curl command - 
```
 curl --request POST 'http://localhost:3000/forms/chromium/convert/html' --header 'Gotenberg-Trace: debug' --form 'files=@"index.html"' -o my.pdf
```

This always outputs a regular white-background pdf.
Please help if you can

**Comment by shimonmagal** [comment]
I checked this again now - and it appears it is working in gotenberg:6 but in 7.

**Comment by gulien** [comment]
Hello @shimonmagal,

I think you're missing the form field `printBackground`:

```
printBackground   bool    Print the background graphics (default false)
```

Setting it to true should do the trick.

**Reply by NiklasBr** [reply]
@gulien this works if you have no margins or headers/footers, but once you do the result is this:

```html

<!DOCTYPE html>
<html>
<head>
    <style>
        @media print {
            @page {
                margin: 22mm 15mm 35mm 15mm;
                background-color: teal !important;
            }
            html, body, #content, #header, #footer, .page {
                background-color: teal !important;
                -webkit-print-color-adjust: exact !important;
            }
        }
    </style>
</head>
<body>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>Lipsum....</p>
    <p>And so on....</p>
</body>
</html>
```

https://user-images.githubusercontent.com/279826/204599408-a66c130d-ee80-428b-a9c3-ff38010ca441.png

**Comment by shimonmagal** [comment]
Sure, thanks all. Yes, now I am able to print it but it is only printed to the first page.
I am using background-repeat and everything.

I need to be set on every page. Any examples of that?

**Reply by NiklasBr** [reply]
I tried reporting this issue to the Puppeteer repo but was shut down, apparently it works as intended?

https://github.com/puppeteer/puppeteer/issues/9341
